apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -178,14 +182,14 @@ export class ProjectChangeAnalyzer {
       const owningProject: RushConfigurationProject | undefined =
         this._rushConfiguration.findProjectForPosixRelativePath(filePath);
       if (owningProject) {
-        // At this point, `filePath` is guaranteed to start with `projectRelativeFolder`, so
-        // we can safely slice off the first N characters to get the file path relative to the
-        // root of the `owningProject`.
-        const relativePath: string = filePath.slice(owningProject.projectRelativeFolder.length + 1);
-        const ignoreMatcher: Ignore | undefined = ignoreMatcherForProject.get(owningProject.packageName);
-        if (!ignoreMatcher || !ignoreMatcher.ignores(relativePath)) {
-          projectHashDeps.get(owningProject.packageName)!.set(filePath, fileHash);
+        let owningProjectHashDeps: Map<string, string> | undefined = projectHashDeps.get(
+          owningProject.packageName
+        );
+        if (!owningProjectHashDeps) {
+          owningProjectHashDeps = new Map<string, string>();
+          projectHashDeps.set(owningProject.packageName, owningProjectHashDeps);
         }
+        owningProjectHashDeps!.set(filePath, fileHash);
```

Comment:
Why is the non-null assertion required here?

Reply:
Good catch. Was a leftover